firebase_admin/dynamic_links.py

```diff
@@ -94,54 +111,45 @@ def event_stats(self):
-        if isinstance(platform, six.string_types) and platform in self._platforms.keys():
-            raise ValueError(('Raw string "{}" detected. Use a dynamic_links.PLATFORM_* constant' +
-                              ' or the make_from_strings() method.').format(platform))
-        if not isinstance(platform, six.string_types) or platform not in self._platforms.values():
-            raise ValueError('platform {}, not recognized'.format(platform))
-        self._platform = platform
-
-        if isinstance(event, six.string_types) and event in self._event_types.keys():
-            raise ValueError(('Raw string {} detected. Use one of the dynamic_links.EVENT_TYPES_' +
-                              ' constants, or the make_from_strings() method.').format(event))
-        if not isinstance(event, six.string_types) or event not in self._event_types.values():
-            raise ValueError('event_type {}, not recognized'.format(event))
-        self._event = event
-
-        if not isinstance(count, int) or isinstance(count, bool) or count < 0:
-            raise ValueError('Count: {} must be a non negative int'.format(count))
-        self._count = count
-
-    @classmethod
-    def make_from_strings(cls, platform, event, count):
-        """make_from_strings creates an EventStat object given the appropriate constants. e.g:
-        make_from_strings(platform=PLATFORM_DESKTOP, event=EVENT_TYPE_REDIRECT, count=4)"""
-        return EventStats(cls._platforms[platform],
-                          cls._event_types[event],
-                          int(count))
+    """``EventStat`` is a single stat item containing (platform, event, count)"""
+
+    def __init__(self, **kwargs):
+        """Create new instance of EventStats(platform, event, count)
+           The input values are the strings returned by the REST call.
+           The internal values stored in the ``EventStats`` object are
+           the package constants named at the start of this package."""
+        required = {'platform', 'event', 'count'}
+        params = set(kwargs.keys())
+        missing = required - params
+        unexpected = params - required
+        if missing:
+            raise ValueError('Missing arguments for EventStats: {}'.format(missing))
+        if unexpected:
+            raise ValueError('Unexpected arguments for EventStats: {}'.format(unexpected))
+
+        platform = kwargs['platform']
+        if not isinstance(platform, six.string_types) or platform not in _platforms.keys():
+            raise ValueError('Invalid Platform value "{}".'.format(platform))
+        self._platform = _platforms[platform]
+
+        event = kwargs['event']
+        if not isinstance(event, six.string_types) or event not in _event_types.keys():
+            raise ValueError('Invalid Event Type value "{}".'.format(event))
+        self._event = _event_types[event]
+
+        count = kwargs['count']
+        if(not ((isinstance(count, six.string_types)    # a string
```

Comment:
Way too defensive here. Does the server return an int here? If so simply check for that (the 2nd part of your `or` statement), followed by `int(count)`. We should handle missing fields, but if a server reports an int field as string, we should not try to mask that.

Reply:
The server returns a quoted int. 
Typically the first line should have been enough.
(the bool guard is a python quirk)
Let's discuss offline.